### DISCUSSION TITLE
How to backup and restore a full system from ZBM?

### DISCUSSION THREAD

**Discussion by neilpanchal** [zbm-dev/zfsbootmenu#635] [Questions] [answered]
I noticed there is an option to take snapshots using Ctrl + S. Is it possible to send this snapshot over ssh to a backup NAS? And also restore it incase of disk failure (which would also destroy ZBM bootloader)?

This is what I am planning to do, chronologically:

1. Install ZFSBootMenu + a debian system.
2. Take a snapshot of the system once a week and send to a backup system over ssh.
3. Now imagine a disk failure where ZBM + debian system are destroyed. The entire disk has failed.
4. Recovery process starts by replacing the disk, installing ZBM + Debian system.
5. ZBM receive and overwrite the new debian system with a previous snapshot.
6. Boot into the restored debian system.

**Comment by classabbyamp** [comment, marked as answer]
the snapshots ZBM shows are just regular zfs snapshots, and can be managed/sent offsite with any zfs snapshotting tool

If you have snapshots as backups, you should be able to boot into ZBM (e.g. using a [portable rescue ZBM image](https://docs.zfsbootmenu.org/en/v2.3.x/general/portable.html)) without installing the OS, `zfs recv` the snapshot you want, and bootstrap the new system that way (this also works for cloning an install to a new machine)

**Reply by neilpanchal** [reply]
Thank you, should I be using the emergency shell to run zfs recv? I noticed apt or ssh isn't installed in the emergency shell and not sure how I can setup networking there.

**Reply by neilpanchal** [reply]
I was able to setup snapshotting/schedule on the debian system itself after booting into it. This is great so the only part I need to figure out is how to restore zfs snapshot into a brand new drive with the help of ZBM.

**Reply by classabbyamp** [reply]
the recovery image provided/built by ZBM contains `dhclient`, which can be used to set up a dhcp connection (https://docs.zfsbootmenu.org/en/v2.3.x/general/binary-releases.html#recovery-images)

otherwise, you could create your own image with networking support (https://docs.zfsbootmenu.org/en/v2.3.x/general/remote-access.html)

**Reply by neilpanchal** [reply]
Beautiful, thank you for the guidance.

**Reply by no-usernames-left** [reply]
I honestly don't see why there are two prebuilt images. Since almost everyone will be downloading and using the prebuilt image, there should be only one prebuilt image which can reasonably do everything needed, including networking, rsync, etc, and that should be the image currently named "recovery".

**Reply by classabbyamp** [reply]
not everyone needs or wants everything in the recovery image

**Reply by neilpanchal** [reply]
It'd be great to get an ISO image since it's easy to use on VMs.

**Reply by neilpanchal** [reply]
@classabbyamp Is there any downside to just using recovery image for everything? I don't mind the slight bloat (that I might need in the future)? It's only a few more utilities. Thanks!

**Reply by classabbyamp** [reply]
not really

**Reply by no-usernames-left** [reply]
> not really

And this is precisely why. If you can spare the extra ≈20 MB, and it might save your ass one day, why have two images? If you are interacting with ZBM in any way that involves a couple of keystrokes and maybe a passphrase, you are probably having a Bad Day, so why limit your options?

> not everyone needs or wants everything in the recovery image

Nobody needs or wants that functionality until the day they do, at which point they probably aren't in a position to easily replace the image.

And if you have both installed, why waste the space given that one is a strict subset of the other?

(Don't get me wrong, I love ZBM.)

**Reply by classabbyamp** [reply]
ok, so do that if you want to. we gave you as user the choice...
